libqtile/layout/floating.py

```diff
@@ -88,19 +86,41 @@ def __init__(self, float_rules=None, no_reposition_match=None, **config):
         the screen.
         """
         Layout.__init__(self, **config)
+        self.add_defaults(Floating.defaults)
         self.clients = []
         self.focused = None
         self.group = None
-        self.float_rules = float_rules or DEFAULT_FLOAT_RULES
         self.no_reposition_match = no_reposition_match
-        self.add_defaults(Floating.defaults)
+
+        if override_default_float_rules:
```

Comment:
Is there a reason to have this flag? It seems sort of clunky.

Reply:
The idea is to provide the user with the possibility to only use her own rules or no floating rules at all. I added this because of issue #1275 .

Reply:
I admit it is sort of clunky. Maybe a shorter name appears less clunky.

Reply:
Can't we just always set the floating rules to whatever the config contains? If it's unspecificed, it'll get set to what's in the default config, and if it's specified, we want to respect the user's settings.

The way we did it before was kind of silly, with an out-of-band configuration thing.

Reply:
Almost all users will want the preset rules because they are so basic. That's why I think it's good to have the user make an extra step to override them. Most users just want to add some specific applications.

Without the override-flag, these basic rules need to be in the users config, so we could add them to the `default_config.py`. But then the behaviour for people with older config files changes.

Reply:
> That's why I think it's good to have the user make an extra step to override them.

They already said they want to override them when they set custom rules. Making them set it again is just silly. Inevitably someone will miss it, and they'll ask me about it, and I'll have to grep through the code and say "oh you have to set this other flag too" :). Let's add them to default_config.py, and break older users. As long as there's a clear note in `/CHANGELOG` about how to forward port things, I think it's an improvement.

Reply:
It's not a hard break anyway. And probably not affecting too many.

Reply:
Also, if users want to just extend the rules, they can simply do something in the lines of:

```python
floating = Floating()
floating.float_rules.append({...})
```

or if we prefer your defaultdict API:

```
floating = Floating()
floating.float_rules["wmtype"].append(...)
```

This would requires to use lists rather than sets though, as sets aren't extensible.